geoutils/georaster.py

```diff
@@ -844,22 +880,25 @@ def reproject(
             # Warning: this will not work for multiple bands with different dtypes
             dtype = self.dtypes[0]
 
-        if nodata is None:
-            nodata = self.nodata
-            # If no data was set, need to set one by default, in case reprojection is done outside original bounds
-            # Otherwise, output nodata will be 99999 by default which will not work as expected for uint8 data.
-            if nodata is None:
-                if not silent:
-                    warnings.warn("No nodata set, will use 0")
-                nodata = 0
+        # Set source nodata if provided
+        if src_nodata is None:
+            src_nodata = self.nodata
+
+        # Set destination nodata if provided. This is needed in areas not covered by the input data.
+        # If None, will use GeoUtils' default, as rasterio's default is unknown, hence cannot be handled properly.
+        if dst_nodata is None:
+            dst_nodata = self.nodata
+            if dst_nodata is None:
+                dst_nodata = default_ndv(dtype)
```

Comment:
Is dtype really always a string here? Can't it be a `np.dtype` as well? I'm asking because your `default_ndv` function seems to require a `str`.

Reply:
Mmmh good point. The argument of reproject is flagged as a np.dtype. Do you know an easy way to convert from one to another?

Reply:
Fixed in ae955e5